packages/core/src/connectors/index.ts

```diff
@@ -92,23 +92,39 @@ const loadConnectors = async () => {
 
 export const getLogtoConnectors = async (): Promise<LogtoConnector[]> => {
   const connectors = await findAllConnectors();
-  const connectorMap = new Map(connectors.map((connector) => [connector.id, connector]));
 
-  const logtoConnectors = await loadConnectors();
+  const loadConnectorPackages = await loadConnectors();
+  const loadConnectorPackageMap = new Map(
+    loadConnectorPackages.map((loadConnectorPackage) => [
+      loadConnectorPackage.metadata.id,
+      loadConnectorPackage,
+    ])
+  );
 
-  return logtoConnectors.map((element) => {
-    const { id } = element.metadata;
-    const connector = connectorMap.get(id);
+  return Promise.all(
+    connectors.map(async (connector) => {
+      const {
+        connectorId,
+        metadata: { target, logo, logoDark },
+      } = connector;
+      const loadConnectorPackage = loadConnectorPackageMap.get(connectorId);
 
-    if (!connector) {
-      throw new RequestError({ code: 'entity.not_found', id, status: 404 });
-    }
+      if (!loadConnectorPackage) {
+        throw new RequestError({ code: 'entity.not_found', connectorId, status: 404 });
+      }
 
-    return {
-      ...element,
-      dbEntry: connector,
-    };
-  });
+      return {
+        ...loadConnectorPackage,
+        metadata: {
+          ...loadConnectorPackage.metadata,
+          target: target ?? loadConnectorPackage.metadata.target,
```

Comment:
`...connector.metadata`?

Reply:
was concerning whether `connector.metadata.logoDark` can be null and would overwrite the existing logoDark value.

Reply:
Ah, maybe this is expected.
`...connector.metadata` should work fine.